external/fv3fit/fv3fit/keras/_models/shared/training_loop.py

```diff
@@ -94,7 +87,13 @@ def fit_loop(
             callbacks=[EpochCallback(func) for func in callbacks], epochs=self.epochs
         )
         if self.in_memory:
-            Xy_fit, validation_fit = _tfdataset_to_tensor_sequence(Xy, validation_data)
+            Xy_fit = _tfdataset_to_tensor_sequence(Xy)
```

Comment:
Nice job moving the iffy logic up the call stack.

Reply:
i recently experience this too. This is why the `Maybe` class exists in haskell btw:
```
Prelude> times_2 x = 2 * x
Prelude> times_2 <$> Just 2
Just 4
Prelude> times_2 <$> Nothing
Nothing
```
Makes it so much easier to compose. This adds 1, multiples by 2, and then squares:
```
Prelude> (^2) <$> (*2) <$> (+1) <$> Nothing
Nothing
Prelude> (^2) <$> (*2) <$> (+1) <$> Just 1
Just 16
```